### DISCUSSION TITLE
How to change the color/position of Graphics Geometry between renders

### DISCUSSION THREAD

**Discussion by MattRCole** [pixijs/pixijs#8125] [Q&A] [answered]
Hi!

I'm building a clone of the popular 0 player game: `Multiply or Release`. Currently, I'm working on the game grid.

Here's a screen shot taken from someone's working [`Multiply or Release`](https://www.youtube.com/watch?v=FcqsnGymYcQ])'s game grid:

<img width="435" alt="example-grid" src="https://user-images.githubusercontent.com/24833910/150480351-72cd4b98-7d6b-4bdd-b427-0a064a60e2cc.png">

Here are the graphical constraints of the grid:
1. There can be up to several hundreds of balls moving around on screen
1. Under certain conditions, when a ball collides with a cell, that cell will change colors. 
    - Example from the screen shot: the balls in the orange/yellow section of the grid are moving towards, and about to collide with, some blue cells. When the balls collide with the blue cells, any cell that is hit will change colors from blue to orange

I've gotten something similar to the screen shot working, but I'd like to improve its render times. Currently I have 2 instances of the `Graphics` class: one for balls, one for the grid, and I am only updating the grid when a collision is detected. This is way more efficient that what I was doing before (`clear`ing out the gird `Graphics` instance's  geometry every frame, re-populating it and then drawing it)

This brings me to my 2 questions: 

1. Is it possible to translate/change the position of a specific piece of geometry in a `Graphics` instance? If it is, I could simply translate the position of a ball instead of clearing it's geometry and redrawing it.
1. Is it possible to change the fill color of a specific piece of geometry in a `Graphics` instances' geometry list? If it is, I could create the grid once and update cell fill colors as necessary.

**Comment by domis86** [comment]
>  Currently I have 2 instances of the Graphics class: one for balls, one for the grid

* Question about "balls":  do you have "1 Graphics per ball" ( so if there is 8 balls then there is 8 Graphics etc) or "1 Graphics to draw all balls in it" ?
* Question about "grid":  do you have "1 Graphics per cell" or "1 Graphics to draw all cells in it" ?

**Reply by MattRCole** [reply]
- 1 Graphics to draw all balls in it
- 1 Graphics to draw all cells in it

**Reply by domis86** [reply]
Ok - so that is not very good solution: you should try using 1 Graphics per ball and 1 Graphics per cell. This way you will have better control over individual balls and cells, and the drawing performance should be better.

But even better solution is what @bigtimebuddy said in comment below: https://github.com/pixijs/pixijs/discussions/8125#discussioncomment-2014478

**Comment by bigtimebuddy** [comment, marked as answer]
The answer to both questions is basically no. This is not the intended use of graphics which are largely designed to be static. Turning all the internal shapes into geometry (triangulating) is expensive performance-wise, so clearing every frame is going to come at a steep cost.

You will get way better performance using sprites with shared textures OR multiple (hundreds) graphics with shared geometry (to avoid re-triangulation). Since many of the squares are the same you may consider ParticleContainer, which is well optimized for this type of thing.

Good luck!

**Reply by MattRCole** [reply]
Awesome, I will look into ParticleContainer!
